components/sup/src/manager/service/supervisor.rs

```diff
@@ -198,13 +195,10 @@ impl Supervisor {
                     // of this comment.
                     Ok(v) if v > 14227 => pkg.svc_user.clone(),
                     Ok(_) => legacy_user,
-                    Err(err @ Timeout) => {
+                    Err(err) => {
                         error!("Error getting version from launcher: {:?}", err);
                         legacy_user
                     }
```

Comment:
On error, this assumes the launcher is "old", correct? This means that if it is really a "new" launcher we will get an error when we try to login as a different user.

It looks like the `UnknownMessage` response was [added somewhat recently](https://github.com/habitat-sh/habitat/blob/b48f8f443dc690e24dbf8adf5725fd84d04f97f1/components/launcher/src/server.rs#L510). This means there are really three kinds of launchers:

1. Only respond with `Timeout` on version check
2. Respond with `UnknownMessage` on version check unless there truly was a timeout
3. Respond with `Version` on version check unless there truly was a timeout or an `InvalidVersionNumber`

I agree it seems like we have to assume we are dealing with a legacy launcher on `UnknownMessage` or `Timeout` and just log an error message, but it seems like some errors (eg `InvalidVersionNumber`) it would be more helpful to return the error. Can we even get an `InvalidVersionNumer` error unless it is a new launcher?

Reply:
It may also be beneficial to us debugging current installs to capture more information about the nature of the error.

Reply:
I now explicitly handle `Timeout` and `UnknownMessage` and bubble up any other error.